File: Expense_backend/src/main/java/com/project/ExpenseTracker/services/income/IncomeService.java
package com.project.ExpenseTracker.services.income;


import com.project.ExpenseTracker.dto.IncomeDTO;
import com.project.ExpenseTracker.entity.Income;

import java.util.List;

public interface IncomeService {
    Income postIncome(IncomeDTO incomeDTO);
    List<IncomeDTO> getAllIncomes();
    Income updateIncome(Long id, IncomeDTO incomeDTO);
    IncomeDTO getIncomeById(Long id);
    void deleteIncome(Long id);
}

File: Expense_frontend/expenseTrackerWeb/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/"

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  constructor(private httpClient : HttpClient) { }

  postIncome(incomeDTO: any): Observable<any> {
    return this.httpClient.post(BASIC_URL + "api/income", incomeDTO);
  }

  getAllIncome(): Observable<any> {
    return this.httpClient.get(BASIC_URL + "api/income/all");
  }

  getIncomeByID(id: number): Observable<any> {
    return this.httpClient.get(BASIC_URL + `api/income/${id}`);
  }

  updateIncome(id: number, incomeDTO: any): Observable<any> {
    return this.httpClient.put(BASIC_URL + `api/income/${id}`, incomeDTO);
  }

  deleteIncome(id: number): Observable<any> {
    return this.httpClient.delete(BASIC_URL + `api/income/${id}`);
  }
}
